### DISCUSSION TITLE
#if conditionals doesn't highlight code properly

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#17940] [Q&A] [answered]
Parts of my code should only execute on Windows, for which I use the HAS_UNO conditional. However, in Visual Studio, the code appears to be disabled, even though it actually runs.

Visual Studio should enable or disable the code depending on whether it is for Windows or another platform.

![image](https://github.com/user-attachments/assets/1981dc1e-4dba-4e91-8170-8dd5c93338a0)

**Comment by morning4coffe-dev** [comment]
Hey @mcNets,

This actually depends on the platform you have selected at the top of the editor and it is not related to the currently targeted platform. Try switching between platforms there to see if it works:
![image](https://github.com/user-attachments/assets/ea8aa686-2041-42b8-932d-f9a800606213)

**Reply by mcNets** [reply]
Hi @morning4coffe-dev 

Shouldn't ```WinAppSDK Unpackaged``` highlight the code inside ```#if !HAS_UNO```? That in fact is executed.

**Reply by morning4coffe-dev** [reply, marked as answer]
I don’t think so, as that’s not how VS handles this—it only highlights when you change the editor platform:
![image](https://github.com/user-attachments/assets/82807a90-479a-4eb6-b9ad-6ce522ba5881)
 
Should the editor platform change when you change the target platform? Maybe, but I don't think that’s currently supported by our SingleProject implementation in VS. 🤔
